clients/src/main/resources/common/message/SyncGroupRequest.json

```diff
@@ -18,18 +18,23 @@
   "type": "request",
   "name": "SyncGroupRequest",
   // Versions 1 and 2 are the same as version 0.
-  "validVersions": "0-2",
+  // Starting from version 3, we add a new field called groupInstanceId to indicate member identity across restarts.
+  "validVersions": "0-3",
   "fields": [
     { "name": "GroupId", "type": "string", "versions": "0+", "entityType": "groupId",
       "about": "The unique group identifier." },
     { "name": "GenerationId", "type": "int32", "versions": "0+",
       "about": "The generation of the group." },
     { "name": "MemberId", "type": "string", "versions": "0+",
       "about": "The member ID assigned by the group." },
+    { "name": "GroupInstanceId", "type": "string", "versions": "3+", "nullableVersions": "3+",
+      "about": "The unique identifier of the consumer instance provided by end user." },
     { "name": "Assignments", "type": "[]SyncGroupRequestAssignment", "versions": "0+",
       "about": "Each assignment.", "fields": [
       { "name": "MemberId", "type": "string", "versions": "0+",
         "about": "The ID of the member to assign." },
+      { "name": "GroupInstanceId", "type": "string", "versions": "3+", "nullableVersions": "3+",
```

Comment:
Hmm.. Why do we need to send the instance IDs of the individual members?

Reply:
Good question, I was thinking it would be useful for partition assignments in long term, but a second thought thinks it's unnecessary. Will remove it.